src/tests/system/tests/test_sss_override.py

```diff
@@ -0,0 +1,263 @@
+        2. Start SSSD
+    :steps:
+        1. Create local override "group1" and name it "o-group1"
+        2. Restart SSSD, this is necessary to enable local overrides
+        3. Query groups by the local override name
+        4. Override the GID for the group "group1"
+        5. Query groups by the override name
+    :expectedresults:
+        1. Group local override is created
+        2. SSSD has restarted successfully
+        3. Group is found by the overriden name "o-group1"
+        4. Local override POSIX attribute updated
+        5. Group is found by the overriden name "o-group1" and GID changed
+    :customerscenario: False
+    :requirement: True
+    """
+    provider.group("group1").add(gid=999999)
+    client.sssd.domain["ldap_id_mapping"] = "False"
+    client.sssd.config_apply()
+    client.sssd.start()
+
+    group = client.sss_override.group("group1")
+
+    group.add(name="o-group1")
+    client.sssd.restart()
+
+    result = client.tools.getent.group("group1")
+    assert result is not None
+    assert result.gid == 999999
+    assert client.tools.getent.group("o-group1")
+    group.add(name="o-group1", gid=888888)
+
+    assert client.tools.getent.group("group1")
+    assert client.tools.getent.group("o-group1")
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+def test_sss_overrides__root_uid_gid_cannot_be_used(client: Client, provider: GenericProvider):
+    """
+    :title: Root user UID/GID cannot be overridden
+    :setup:
+        1. Create POSIX user "user1" with standard POSIX attributes defined
+        2. Start SSSD
+    :steps:
+        1. Create local override "root" for user1 and set UID/GID to 0
+        2. Restart SSSD, this is necessary to enable local overrides
+        3. Query the root user
+        4. Query the root user and use sss as the service
+        5. Query the POSIX user that is overridden to the root user
+    :expectedresults:
+        1. Local override is created
+        2. SSSD has restarted successfully
+        3. The root user UID/GID has not been modified
+        4. The override has no UID/GID attribute
+        5. The POSIX user UID/GID has not been changed
+    :customerscenario: False
+    :requirement: True
+    """
+    provider.user("user1").add(
+        uid=999011, gid=999011, home="/home/user1", gecos="user", shell="/bin/bash", password="Secret123"
+    )
+    client.sssd.domain["ldap_id_mapping"] = "False"
+    client.sssd.config_apply()
+    client.sssd.start()
+
+    user = client.sss_override.user("user1")
+
+    user.add(name="root", uid=0, gid=0)
+    client.sssd.restart()
+
+    result = client.tools.getent.passwd("root")
+    assert result.uid == 0
+    assert result.gid == 0
+
+    result = client.tools.getent.passwd("root", service="sss")
+    assert hasattr(result, "uid") is False
+    assert hasattr(result, "gid") is False
```

Comment:
Shouldn't this say `assert result is None` since the user is not found?

Reply:
If result is None, hasattr check is redundant. None has no attributes. So these two assertions will never fail.

Reply:
This is still not resolved, please remove hasattr checks as it can never be true.

`assert result is None` ensures that result does not have any attributes because it is `None` so `hasattr` will never find the attribute therefore this check is meaningless. And if the root user is found then `assert result is None` will fail and the test execution will finished with FAILED result, we never reach `hasattr` checks.